source/particle/property/crystal_preferred_orientation.cc

```diff
@@ -0,0 +1,1160 @@
+                                         "Spin tensor, D-Rex 2004."))
+                }
+
+              const std::string temp_advection_method = prm.get("Property advection method");
+              if (temp_advection_method == "Forward Euler")
+                {
+                  advection_method = AdvectionMethod::forward_euler;
+                }
+              else if (temp_advection_method == "Backward Euler")
+                {
+                  advection_method = AdvectionMethod::backward_euler;
+                }
+              else if (temp_advection_method == "Crank-Nicolson")
+                {
+                  advection_method = AdvectionMethod::crank_nicolson;
+                }
+              else
+                {
+                  AssertThrow(false, ExcMessage("particle property advection method not found: \"" + temp_advection_method + "\""));
+                }
+
+              prm.enter_subsection("Initial grains");
+              {
+                const std::string model_name = prm.get("Model name");
+                AssertThrow(model_name == "Uniform grains and Random Uniform rotations",
+                            ExcMessage("No model named " + model_name + "for CPO particle property initialization. "
+                                       + "Only the model \"Uniform grains and Random Uniform rotations\" is available."));
+                prm.enter_subsection("Uniform grains and Random Uniform rotations");
+                {
+                  const std::vector<std::string> temp_deformation_type_selector = dealii::Utilities::split_string_list(prm.get("Minerals"));
+                  n_minerals = temp_deformation_type_selector.size();
+                  deformation_type_selector.resize(n_minerals);
+
+                  for (size_t mineral_i = 0; mineral_i < n_minerals; mineral_i++)
+                    {
+                      if (temp_deformation_type_selector[mineral_i] == "Passive")
+                        {
+                          deformation_type_selector[mineral_i] = DeformationTypeSelector::passive;
+                        }
+                      else
+                        {
+                          AssertThrow(false,
+                                      ExcMessage("The fabric needs to be assigned one of the following comma-delimited values: Olivine: Karato 2008, "
+                                                 "Olivine: A-fabric,Olivine: B-fabric,Olivine: C-fabric,Olivine: D-fabric,"
+                                                 "Olivine: E-fabric, Enstatite and Passive."))
+                        }
+                    }
+
+                  volume_fractions_minerals = Utilities::string_to_double(dealii::Utilities::split_string_list(prm.get("Volume fractions minerals")));
+                  double volume_fractions_minerals_sum = 0;
+                  for (auto &&fraction : volume_fractions_minerals)
```

Comment:
No need for the && when iterating over a vector of doubles. Actually slightly surprised this was not iterating over the memory addresses of volume_fractions_minerals instead of the values. Wouldnt it be of type `double *` in the current form?

```suggestion
                  for (auto fraction : volume_fractions_minerals)
```